### DISCUSSION TITLE
Demo UI Ideas

### DISCUSSION THREAD

**Discussion by Magnogen** [aseaday/MarkovJunior.js#5] [Ideas]
I was playing with the organisation of the two main features, and also added more of a cohesive colour palette. What do you think?

![code on the right](https://user-images.githubusercontent.com/25611707/173228255-47deb04a-5175-4d42-8e16-e18fb395bf2b.png)

of course, it could quite as easily be swapped over

![code on the left](https://user-images.githubusercontent.com/25611707/173228280-7a5085a3-c1a1-4adb-919a-2c1afe1646d9.png)

or just about anything. What do you prefer?

Also, syntax highlighting. It's just a matter of putting a `<span>` around the text you want styled, and styling that. Easily adjustable.

![syntax highlight example](https://user-images.githubusercontent.com/25611707/173228501-7e9be7c3-c132-4db2-b56e-ad203ef12398.png)

**Comment by lalawuu** [comment]
I could provide a xml formatter and syntax highlight utils.

**Comment by x2nie** [comment]
hi ! yeah !  I went to it by reading the issue/question.
Now, about YML / Python,
what if be something like this:
### Sample syntax:
```python
# BasicKeys.xml
sequence (BDIRGYNOKPFCU, origin=True):
    all (DB* -> *ID) # "draw a grid"

    one (D -> G, limit=1) # "put an end"

    # put a start far from the end
    one: 
        rule (D -> R limit=1)
        field (for="R" from="G" on="RID")
    
    one (RID -> *NO, limit=1)

    # draw a non-optimal path from start to end
    one (temperature=2.0): 
        rule (OID -> YNO)
        rule (OIG -> YNG)
        field (for=O, to=G, on="GIDO")
    
    one (RNY -> KFR, limit=1)

    markov:
        one (UID -> CCU)
        one (RID -> OCU)
        one:
            rule (RNY -> FFR)
            rule (ONY -> FPR)
            rule (RNG -> FFG)
            rule (ONG -> FPG)

    all (CID -> *CC)
    all (I -> B)
```
